satpy/readers/seviri_l1b_native.py

```diff
@@ -526,138 +518,134 @@ def calibrate(self, data, dataset_id):
-            line_start = south - final_shape[0] + data_shape[0]
-            line_end = line_start + nlines - 1
-
-            # Pad data in east-west direction
-            data_window = pad_data_ew(dataset[line_start - 1:line_end, :].data,
-                                      (nlines, final_shape[1]),
-                                      east, west)
+        """Pad FES HRV data to full disk with empty pixels.
 
-            data_list.append(data_window)
-
-        padded_data = da.vstack(data_list)
-
-        return xr.DataArray(padded_data, dims=('y', 'x'))
-
-    def _pad_rss_roi_data(self, dataset_id, dataset):
-        """Pad RSS and ROI data to full disk with empty pixels."""
-        data_shape = dataset.shape
+        The HRV channel in FES mode consists of data from two windows, covering all 11136 lines. Data from these
+        two windows are read, padded horizontally and finally stacked vertically to form a full disk array.
+        """
+        data_list = []
+        for south_bound, north_bound, east_bound, west_bound in zip(*self._img_bounds.values()):
+            nlines = north_bound - south_bound + 1
+            padded_data = pad_data_horizontally(dataset[south_bound - 1:north_bound, :].data,
+                                                (nlines, self._final_shape[1]), east_bound, west_bound)
+            data_list.append(padded_data)
 
-        if dataset_id['name'] == 'HRV':
-            final_shape = (HRV_NUM_LINES, HRV_NUM_COLUMNS)
-        else:
-            final_shape = (VISIR_NUM_LINES, VISIR_NUM_COLUMNS)
+        return xr.DataArray(da.vstack(data_list), dims=('y', 'x'))
 
-        # The HRV channel in RSS mode consists of data from one window ('Lower'). If we're dealing with
-        # such data, we need to use the actual navigation parameters in order to pad the data correctly.
-        # Otherwise, we use the selected navigation parameters.
+    def _pad_rss_roi_data(self, dataset):
+        """Pad RSS and ROI data to full disk with empty pixels.
 
-        # Pad data in east-west direction
-        east, west = self._boundary['east'][0], self._boundary['west'][0]
-        padded_data = pad_data_ew(dataset.data, (data_shape[0], final_shape[1]), east, west)
+        The VISIR and HRV channels in RSS and ROI mode consists of data from one single window/area. Hence,
+        we start by extracting the first (only) elements of the image boundary lists. We then pad the data
+        both horizontally and vertically to form a full disk array.
+        """
+        south_bound, north_bound, east_bound, west_bound = [v[0] for v in self._img_bounds.values()]
+        nlines = north_bound - south_bound + 1
 
-        # Pad data in south-north direction
-        south, north = self._boundary['south'][0], self._boundary['north'][0]
-        padded_data = pad_data_sn(padded_data, final_shape, south, north)
+        padded_data = pad_data_horizontally(dataset.data, (nlines, self._final_shape[1]), east_bound, west_bound)
```

Comment:
If I'm reading this correctly, these 3 lines could (almost) be replaced with
```
padded_data = self._pad_fes_hrv_data(dataset)
```
In which case `_pad_fes_hrv_data` should be renamed

(yeah, sorry, I don't like duplicated code :) )

Reply:
I found another way to do the p[adding more concise, but for this solution the bots now complain about possibly undefined variables. Is that acceptable? Because in practice I don't see how they would be undefined. But it's Friday evening and I might very well miss something ;)